nixos/modules/services/web-apps/slskd.nix

```diff
@@ -0,0 +1,211 @@
+{ lib, pkgs, config, ... }:
+
+let
+  settingsFormat = pkgs.formats.yaml {};
+in {
+  options.services.slskd = with lib; with types; {
+    enable = mkEnableOption "enable slskd";
+
+    rotateLogs = mkEnableOption "enable an unit and timer that will rotate logs in /var/slskd/logs";
+
+    package = mkPackageOptionMD pkgs "slskd" { };
+
+    nginx = mkOption {
+      description = lib.mdDoc "options for nginx";
+      example = {
+        enable = true;
+        domain = "example.com";
+        contextPath = "/slskd";
+      };
+      type = submodule ({name, config, ...}: {
+        options = {
+          enable = mkEnableOption "enable nginx as a reverse proxy";
+
+          domainName = mkOption {
+            type = str;
+            description = "Domain you want to use";
+          };
+          contextPath = mkOption {
+            type = types.path;
+            default = "/";
+            description = lib.mdDoc ''
+              The context path, i.e., the last part of the slskd
+              URL. Typically '/' or '/slskd'. Default '/'
+            '';
+          };
+        };
+      });
+    };
+
+    environmentFile = mkOption {
+      type = path;
+      description = ''
+        Path to a file containing secrets.
+        It must at least contain the variable `SLSKD_SLSK_PASSWORD`
+      '';
+    };
+
+    openFirewall = mkOption {
+      type = bool;
+      description = ''
+        Whether to open the firewall for services.slskd.settings.listen_port";
+      '';
+      default = false;
+    };
+
+    settings = mkOption {
+      description = lib.mdDoc ''
+        Configuration for slskd, see
+        [available options](https://github.com/slskd/slskd/blob/master/docs/config.md)
+        `APP_DIR` is set to /var/lib/slskd, where default download & incomplete directories,
```

Comment:
Is there a reason the application directory isn't configurable?

Reply:
As for `slskd` user, I preferred to do things incrementally, and let someone with particular needs like you make a new PR. I encourage you to do so :)

On my setup, which serves a lot of files, `APP_DIR/data` weights ~250 MB. I didn't think of which use cases would need this somewhere else, so I let this not configurable. However I encourage you to add more flexibility to the module.

I don't have time to address your issues in the following days. I will may be address them in ~10 days, but I'll welcome a PR instead.

Reply:
That's alright, it's not a particular issue for me. I just wondered why it was referred like a variable when it wasn't (yet). I might make a PR for these fixes this week, it was just easiest for me to share my thoughts here beforehand.

Reply:
Cool. Oh, right, APP_DIR is an environment variable [read by slskd](https://github.com/slskd/slskd/blob/master/docs/config.md#application-directory-configuration), but it's hardcoded, not "variabilized", in the nix module.

It's nice to express your thoughts before making a PR, thanks. If you *PR* and want to add yourself to the maintainers list, you're welcome to do so.